src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -195,14 +195,18 @@ private function createRoleHierarchy(array $config, ContainerBuilder $container)
     private function createAuthorization(array $config, ContainerBuilder $container)
     {
         foreach ($config['access_control'] as $access) {
-            $matcher = $this->createRequestMatcher(
-                $container,
-                $access['path'],
-                $access['host'],
-                $access['port'],
-                $access['methods'],
-                $access['ips']
-            );
+            if ($access['request_matcher']) {
+                $matcher = new Reference($access['request_matcher']);
+            } else {
+                $matcher = $this->createRequestMatcher(
+                    $container,
+                    $access['path'],
+                    $access['host'],
+                    $access['port'],
+                    $access['methods'],
+                    $access['ips']
```

Comment:
Should we invalidate the configuration if the app specifies the `request_matcher` option _and_ one of those?

Reply:
If it were to be done, I'd need guidance as to how to do it correctly :)

Reply:
👍 I think you can borrow something from https://github.com/symfony/symfony/blob/6691292b70839dcc46c45a2cb454419e21038dbe/src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php#L80-L83

Reply:
Added, even if this invalidation is not present when building the RequestMatcher for the firewall